src/components/inputs/SearchSelect.jsx

```diff
@@ -0,0 +1,427 @@
+    }
+  },
+
+  _onFilterInputBlur(e) {
+    this._cancelBlurInterval();
+    this._blurInterval = setTimeout(() => {
+      this._clearFilterValue();
+      this._closeList();
+    }, 100);
+  },
+
+  _onFilterInputKeyDown(e) {
+    switch (e.which) {
+    case KEY_CODES.TAB:
+    case KEY_CODES.ENTER:
+      this._selectOption(this.state.suggestedOption);
+      break;
+    case KEY_CODES.ARROW_UP:
+      this._suggestPreviousOption(this.state.filterValue, this.state.suggestedOption);
+      break;
+    case KEY_CODES.ARROW_DOWN:
+      this._suggestNextOption(this.state.filterValue, this.state.suggestedOption);
+      break;
+    }
+  },
+
+  _onGroupClick(group, e) {
+    this._cancelBlurInterval();
+    this._toggleGroupOpenState(group);
+  },
+
+  _onOptionSelect(option, e) {
+    this._selectOption(option);
+  },
+
+  //Renders
+
+  render() {
+    let {
+      className,
+      placeHolder,
+      valueLink: {value},
+    } = this.props;
+    let { isFocused, filterValue } = this.state;
+    let filteredOptions = this._getFilteredOptions(filterValue);
+
+    return (
+      <div
+        className={classNames('Select', `Select--${isFocused ? 'opened' : 'closed'}`, className)}
+        onMouseLeave={this._onBlur}
+      >
+        <div className="Select-input"
+          onClick={this._onSelectValueClick}
+        >
+          <input
+            className={classNames('Select-filterInput', !filterValue && 'Select-filterInput-isEmpty')}
+            type="text"
+            ref="searchInput"
+            value={filterValue}
+            onFocus={this._onFocus}
+            onBlur={this._onFilterInputBlur}
+            onChange={this._onFilterInputChange}
+            onKeyDown={this._onFilterInputKeyDown}
+          />
+          <span className="Select-valueSpan">{value ? value.label : ''}</span>
+          {!filterValue && !value &&
+            <span className="Select-placeholder">{placeHolder}</span>
+          }
+        </div>
+        <div
+          className="Select-optionsList"
+          onMouseEnter={this._onFocus}
+        >
+          {_.map(filteredOptions, (option, i) => {
+            let render = option.isGroup ? this._renderGroup : this._renderOption;
+            return render(option, i);
+          })}
+        </div>
+      </div>
+    );
+  },
+
+  _renderGroup(group, key) {
+    let isOpened = this._isGroupOpen(this.state.openGroupsId, group);
+    return (
+      <div
+        className={classNames("Select-group", `Select-group--${isOpened ? 'opened' : 'closed'}`)}
+        key={key}
+      >
+        <div
+          className="Select-groupHeader"
+          onClick={this._onGroupClick.bind(null, group)}
+        >
+          <i className="Select-groupHeaderIcon" />
+          <span className="Select-groupheaderLabel">{group.label}</span>
+        </div>
+        <div className="Select-groupOptions">
+          {_.map(group.options, this._renderOption)}
+        </div>
+      </div>
+    );
+  },
```

Comment:
Have tried to but there is 2 cons:
- It duplicate the code to render an option.
- It doesn't really move a lot of code as all logic (groupOpen state, ...) need to be in top level component.